### ISSUE TITLE
Add a Cats extension module

### ISSUE THREAD

**Issue by rcavalcanti** [strongtyped/fun-cqrs#61]
The goal is to have an extension module that we can use Validated or Xor as return time for command handlers.

Things that need to be added
- It should be possible to configure a Backend to use another Interpreter
- The Cats extension must provide a its own Backend for testing

**Comment by patricknoir** [comment]
Hi Renato, given that Cats is deprecating Xor in favour of the new Either in scala 2.12 (right biased) do you see still see a value to provide support for Xor?

**Comment by rcavalcanti** [comment]
Good point. Scala 2.12 is almost there so we could hold support for Xor and have it only for the new biased Either.

**Comment by rcavalcanti** [comment]
I have done some progress on this (only analysis and some basic design) and some ideas and issues cames to the surface. 

The main problem I'm facing right now is that whatever effect we choose on the behavior side, it always get transformed to the effect of the backend we use. 

For instance, we have three effects that we can use when defining command handlers: `Future`, `Try` and `Identity`. 

A command handler using `Identity` will return `Identity` when using the `InMemoryBackend` and it will be transformed to `Future` when using the `AkkaBackend`. 

That means the the effects on the behavior definition are not necessarily preserved once executed by the backend. 

I found a good solution to extend `Actions` with new effects: `Validated`, `Either`, `scalaz.Validation`, etc. This will give flexibility when defining command handlers, but whatever effect we choose, it will be discarded once executed. A `Validated` will have to be converted to a `Future`. 

We will end-up with something like:
```scala
Validated[NonEmptyList[String],Events]
class InvalidException(errors: NonEmptyList[String])

// assuming AkkaBackend
val result = aggregateRef ? cmd
// in case of failures, result completed with 
// Failure(InvalidException)
```
Which is awkward. 

The only way of having a `Validated` being returned by a command is by having a `CatsBackend`.

**Comment by rcavalcanti** [comment]
It's now possible to build an extensions (ie: new InvokerDirectives) for Cats.

However, I'm considering making Cats a full citizen in Fun.CQRS. I'm using it in an internal project right now, especially `Validated` and I'm quite happy with the results. 

It's especially handy for fine-grained events.  Whenever we have a `Command` that may emit many `Events`. In that case, if we reject the `Command` we can enumerate all the reasons why is was rejected.
